client/state/stats/chart-tabs/reducer.js

```diff
@@ -30,7 +30,7 @@ const countsReducer = ( state = [], action ) => {
 			// See https://github.com/Automattic/wp-calypso/pull/41441#discussion_r415918092
 			if (
 				action.data.length !== state.length ||
-				! isEqual( first( action.data ).period, first( state ).period )
+				! isEqual( action.data[ 0 ].period, state[ 0 ].period )
```

Comment:
I know this code has been here before, but `data` items are themselves arrays. There is not `.period` field. Response from the `/sites/:site/stats/visits` endpoint looks like this:

<img width="694" alt="Screenshot 2021-04-20 at 14 14 52" src="https://user-images.githubusercontent.com/664258/115394448-119bbf80-a1e3-11eb-91a1-5fc5de8cb8af.png">

Note how the `fields` array specifies what the `data[ x ][ y ]` fields contain.

Reply:
Yes, but the array of fields is transformed to an object at the data layer level - see https://github.com/Automattic/wp-calypso/blob/remove/lodash-first/client/state/data-layer/wpcom/sites/stats/visits/from-api.js#L8 and https://github.com/Automattic/wp-calypso/blob/remove/lodash-first/client/state/stats/lists/utils.js#L239-L269

Reply:
I see, mystery solved 🙂 

Still, `.period` is always a string, never an object. `isEqual` should not be necessary. Merely comparing them with `===` should suffice.

Reply:
Agreed, I prefer skipping it for this PR - there's another `isEqual` usage there that compares objects, and all this isn't directly related to this PR.